profiles/schema.py

```diff
@@ -366,6 +373,40 @@ def mutate(self, info, **kwargs):
         return UpdateProfile(profile=profile)
 
 
+class DeleteProfile(graphene.Mutation):
+    ok = graphene.Boolean()
+
+    class Arguments:
+        id = graphene.ID(required=True)
+
+    @login_required
+    def mutate(self, info, **kwargs):
+        try:
+            profile = Profile.objects.get(pk=from_global_id(kwargs["id"])[1])
+        except Profile.DoesNotExist as e:
+            raise ProfileDoesNotExistError(e)
+
+        # TODO: Should admin users be able to delete profiles?
+        if profile.user != info.context.user:
+            raise PermissionDenied("You do not have permission to perform this action.")
+
+        # TODO: Here we should check whether profile can be
+        #       actually removed. Waiting for spec for this.
+        #       For now just check if user has BERTH.
+        #
+        #       More info:
+        #       https://helsinkisolutionoffice.atlassian.net/browse/OM-248
+        if profile.serviceconnection_set.filter(
+            service__service_type=ServiceType.BERTH
+        ).exists():
+            raise CannotDeleteProfileWhileServiceConnectedError(
+                "Cannot delete profile while service BERTH still connected"
+            )
+
+        result = profile.delete()
+        return DeleteProfile(ok=result[0] > 0)
```

Comment:
maybe:
```python
num_of_deleted_profiles = profile.delete()[0]
return DeleteProfile(ok=num_of_deleted_profiles > 0)
```
or something similar for better readability?

Reply:
It's not number of deleted profiles, it's number of deleted objects in total (including "sub" objects). IMO Django already ruined the readability of this with the implementation of the return value of delete but what the heck, let's go with that 😄